File: VB/02_ParagraphAndText/ReplaceText.vb
Imports Spire.Presentation

Namespace ReplaceText
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			Dim tagValues As New Dictionary(Of String, String)()
			tagValues.Add("Spire.Presentation for .NET", "Spire.PPT")

			'Create an instance of presentation document
			Dim ppt As New Presentation()
			'Load file
			ppt.LoadFromFile("..\..\..\..\..\..\Data\TextTemplate.pptx")

			ReplaceTags(ppt.Slides(0), tagValues)

			'Save the document
			Dim result As String = "ReplaceText.pptx"
			ppt.SaveToFile(result, FileFormat.Pptx2013)
			PresentationDocViewer(result)
		End Sub

		Private Sub PresentationDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub ReplaceTags(ByVal pSlide As ISlide, ByVal TagValues As Dictionary(Of String, String))
			For Each curShape As IShape In pSlide.Shapes
				If TypeOf curShape Is IAutoShape Then
					For Each tp As TextParagraph In (TryCast(curShape, IAutoShape)).TextFrame.Paragraphs
						For Each curKey In TagValues.Keys
							If tp.Text.Contains(curKey) Then
								tp.Text = tp.Text.Replace(curKey, TagValues(curKey))
							End If
						Next curKey
					Next tp
				End If
			Next curShape
		End Sub
	End Class
End Namespace

File: CS/02_ParagraphAndText/ReplaceText.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using Spire.Presentation;

namespace ReplaceText
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            Dictionary<string, string> tagValues = new Dictionary<string, string>();
            tagValues.Add("Spire.Presentation for .NET", "Spire.PPT");

            //Create an instance of presentation document
            Presentation ppt = new Presentation();
            //Load file
            ppt.LoadFromFile(@"..\..\..\..\..\..\Data\TextTemplate.pptx");

            ReplaceTags(ppt.Slides[0], tagValues);

            //Save the document
            string result = "ReplaceText.pptx";
            ppt.SaveToFile(result, FileFormat.Pptx2013);
            PresentationDocViewer(result);
		}
	
		private void PresentationDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void ReplaceTags(ISlide pSlide, Dictionary<string, string> TagValues)
        {
            foreach (IShape curShape in pSlide.Shapes)
            {
                if (curShape is IAutoShape)
                {
                    foreach (TextParagraph tp in (curShape as IAutoShape).TextFrame.Paragraphs)
                    {
                        foreach (var curKey in TagValues.Keys)
                        {
                            if (tp.Text.Contains(curKey))
                            {
                                tp.Text = tp.Text.Replace(curKey, TagValues[curKey]);
                            }
                        }
                    }
                }
            }
        }
    }
}